### DISCUSSION TITLE
An ESLint rule which tells you when you are setting up a loader / action within a component rather than a route

### DISCUSSION THREAD

**Discussion by justinfarrelldev** [remix-run/remix#9602] [Proposals]
Hello,

I just wanted to propose an idea that I thought up a little while ago.

I often find myself accidentally creating a loader / action in a component rather than a route (especially if the route pretty much exclusively contains the component). This, naturally, leads to "server-only functionality used in client" errors. 

Would it be possible to create an ESLint rule to warn if there is an action or a loader present in a client-only file (or files in the components folder)? A rule like this would be pretty useful to warn against this pretty silly mistake I seem to keep making (and I'm sure a lot of people have accidentally done at some point or another).

The only downside I can think of (outside of this maybe being difficult to implement in an ESLint rule) would be that the names "loader" and "action" would not be valid function names within client-side code.

Thank you for your awesome work!

**Comment by sergiodxa** [comment]
The team is working on a `defineRoute` API that will be the only way to create a route with a loader, so just exporting a loader won't work anymore in the future, with this I don't think you will need an ESLint rule anymore.

**Reply by justinfarrelldev** [reply]
Gotcha, makes sense. Thank you for the clarification!
